packages/ra-ui-materialui/src/button/BulkUpdateWithUndoButton.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+import { FC, ReactElement } from 'react';
+import PropTypes from 'prop-types';
+import ActionUpdate from '@material-ui/icons/Update';
+import { fade } from '@material-ui/core/styles/colorManipulator';
+import { makeStyles } from '@material-ui/core/styles';
+import {
+    useUpdateMany,
+    useRefresh,
+    useNotify,
+    useUnselectAll,
+    CRUD_UPDATE_MANY,
+    useResourceContext,
+    useListContext,
+} from 'ra-core';
+
+import Button, { ButtonProps } from './Button';
+import { BulkActionProps } from '../types';
+
+const useStyles = makeStyles(
+    theme => ({
+        updateButton: {
+            color: theme.palette.error.main,
+            '&:hover': {
+                backgroundColor: fade(theme.palette.error.main, 0.12),
+                // Reset on mouse devices
+                '@media (hover: none)': {
+                    backgroundColor: 'transparent',
+                },
+            },
+        },
+    }),
+    { name: 'RaBulkUpdateWithUndoButton' }
+);
+
+const BulkUpdateWithUndoButton: FC<BulkUpdateWithUndoButtonProps> = props => {
+    const {
+        basePath,
+        classes: classesOverride,
+        data,
+        icon,
+        label,
+        onClick,
+        ...rest
+    } = props;
+    const { selectedIds } = useListContext(props);
+    const classes = useStyles(props);
+    const notify = useNotify();
+    const unselectAll = useUnselectAll();
+    const refresh = useRefresh();
+    const resource = useResourceContext(props);
+    const [updateMany, { loading }] = useUpdateMany(
+        resource,
+        selectedIds,
+        data,
+        {
+            action: CRUD_UPDATE_MANY,
+            onSuccess: () => {
```

Comment:
We should also support custom `onSuccess` and `onFailure` prop as functions to customize side effects